secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  RecoveryPhraseStorage.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/10/2022.
+//
+
+import Foundation
+
+/// Zcash implementation of the keychain that is not universal but designed to deliver functionality needed by the wallet itself.
+/// All the APIs should be thread safe according to official doc:
+/// https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
+/// For seamless usage the interface is designed to deal with simple types:
+/// `Seed Phrase: String` - possible to `store` and `retrive` from the keychain. `Delete` operation is also available, use it with caution.
+/// `Birthday: BlockHeight` - possible to `store` and `retrive` from the keychain. `Delete` operation is also available, use it with caution.
+final class RecoveryPhraseStorage {
+    static let `default`:RecoveryPhraseStorage = RecoveryPhraseStorage()
+    
+    enum KeychainError: Error, Equatable {
```

Comment:
Nested types should come before all static properties.
From the docs:
Type
-nested types
-static properties
-constants
-variables
-computed properties
-init methods
-instance methods

Reply:
done